lib/constructs/interface.js

```diff
@@ -26,6 +26,35 @@ Interface.prototype.implements = function (source) {
   this.mixins.push(source);
 };
 
+Interface.prototype.generateNamedPropertiesObject = function () {
+  if (!utils.isGlobal(this.idl)) {
+    return;
+  }
+
+  this.str += `var ${this.name}PropertiesConstructor = function () {
+  throw new TypeError("Illegal constructor");
+}\n`;
+
+  if (this.idl.inheritance) {
+    this.str += `${this.name}PropertiesConstructor.prototype = Object.create(${this.idl.inheritance}.interface.prototype);
+${this.name}PropertiesConstructor.prototype.constructor = ${this.name}PropertiesConstructor;\n`;
+  }
+
+  this.str += `
+var ${this.name}Properties = new Proxy(${this.name}PropertiesConstructor.prototype, {
+  get(target, key) {
+    if (key in this) {
+      return this[key];
+    }
+    return target[key];
+  },
+  set(target, key, value) {
```

Comment:
Per spec we shouldn't use the set trap here. Instead it should be the defineProperty and delete traps both of which should return false.